migrations/1603114571772-rocketchat-to-matrix-messenger.js

```diff
@@ -0,0 +1,93 @@
+/* eslint-disable no-await-in-loop */
+// eslint-disable-next-line no-unused-vars
+const { info, error } = require('../src/logger');
+
+const { connect, close } = require('../src/utils/database');
+
+const { schoolModel } = require('../src/services/school/model');
+
+// How to use more than one schema per collection on mongodb
+// https://stackoverflow.com/questions/14453864/use-more-than-one-schema-per-collection-on-mongodb
+
+// TODO npm run migration-persist and remove this line
+// TODO update seed data and remove this line
+
+module.exports = {
+	up: async function up() {
+		await connect();
+		// ////////////////////////////////////////////////////
+		// Make changes to the database here.
+		// Hint: Access models via this('modelName'), not an imported model to have
+		// access to the correct database connection. Otherwise Mongoose calls never return.
+		const amount = await schoolModel.find({ features: 'rocketChat' }).countDocuments();
+		info(`${amount} schools with rocketChat enabled found. Start migration to Matrix messenger`);
+		const limit = 200;
+		let looped = 0;
+
+		while (looped < amount) {
+			const schools = await schoolModel
+				.find({ features: 'rocketChat' })
+				.sort({
+					updatedAt: 1,
+					createdAt: 1,
+				})
+				.skip(looped)
+				.limit(limit)
+				.lean()
+				.exec();
+			// eslint-disable-next-line no-loop-func
+			await Promise.all(
+				schools.map((school) => {
+					if (!school.features.find((feature) => feature === 'messenger')) {
+						school.features.push('messenger');
+					}
+					return schoolModel.findByIdAndUpdate(school._id, school);
+				})
+			);
+
+			looped += schools.length;
+		}
+		// ////////////////////////////////////////////////////
+		info(`migrated ${looped} schools from rocketChat to Matrix messenger`);
+		await close();
+	},
+
+	down: async function down() {
+		await connect();
+		// ////////////////////////////////////////////////////
+		// Make changes to the database here.
+		// Hint: Access models via this('modelName'), not an imported model to have
+		// access to the correct database connection. Otherwise Mongoose calls never return.
+		const amount = await schoolModel.find({ features: 'messenger' }).countDocuments();
```

Comment:
The current 'down' function will remove all messenger flags (regardless of whether they are added by the migration). Since it is not possible to distinguish them, is it a good idea to provide such a down function?

Reply:
I was asking myself the same question. Both directions are only for the convenience of the admins.

Less code -> Less errors. I guess I should remove the "down"-part.